cpp/include/raft/linalg/detail/rsvd.cuh

```diff
@@ -32,6 +32,97 @@ namespace raft {
 namespace linalg {
 namespace detail {
 
+template <typename math_t>
+void randomized_svd(const raft::handle_t& handle,
+                    math_t* in,
+                    std::size_t n_rows,
+                    std::size_t n_cols,
+                    std::size_t k,
+                    std::size_t p,
+                    std::size_t niters,
+                    math_t* S,
+                    math_t* U,
+                    math_t* V,
+                    bool gen_U,
+                    bool gen_V)
+{
+  common::nvtx::range<common::nvtx::domain::raft> fun_scope(
+    "raft::linalg::randomized_svd(%d, %d, %d)", n_rows, n_cols, k);
+
+  RAFT_EXPECTS(k < std::min(n_rows, n_cols), "k must be < min(n_rows, n_cols)");
+  RAFT_EXPECTS((k + p) < std::min(n_rows, n_cols), "k + p must be < min(n_rows, n_cols)");
+  RAFT_EXPECTS(!gen_U || (U != nullptr), "computation of U vector requested but found nullptr");
+  RAFT_EXPECTS(!gen_V || (V != nullptr), "computation of V vector requested but found nullptr");
+  cudaStream_t stream          = handle.get_stream();
+  cusolverDnHandle_t cusolverH = handle.get_cusolver_dn_handle();
+  cusolverDnParams_t dn_params = nullptr;
+  RAFT_CUSOLVER_TRY(cusolverDnCreateParams(&dn_params));
```

Comment:
We should only be interacting w/ cusolver through the cusolver wrappers so this should be creating the params right in the function itself.